crates/stark-felt/src/lib.rs

```diff
@@ -26,140 +44,201 @@ pub struct FromBytesError;
+        let mut limbs = self.0.representative().limbs;
+        limbs.reverse();
+
+        #[cfg(not(target_pointer_width = "64"))]
+        // Split limbs to adjust to BitArrayStore = [u32; 8]
+        let limbs: [u32; 8] = limbs
+            .map(|n| [(n >> 32) as u32, n as u32])
+            .into_iter()
+            .flatten()
+            .collect::<Vec<u32>>()
+            .try_into()
+            .unwrap();
+
+        BitArray::new(limbs)
     }
 
     /// Converts to little-endian bit representation.
-    pub const fn to_bits_le(&self) -> BitArray<BitArrayStore> {
-        todo!()
+    /// This is as performant as [to_bits_be](Felt::to_bits_be)
+    pub fn to_bits_le(&self) -> BitArray<BitArrayStore> {
+        let limbs = self.0.representative().limbs;
+
+        #[cfg(not(target_pointer_width = "64"))]
+        // Split limbs to adjust to BitArrayStore = [u32; 8]
+        let limbs: [u32; 8] = limbs
+            .map(|n| [n as u32, (n >> 32) as u32])
+            .into_iter()
+            .flatten()
+            .collect::<Vec<u32>>()
+            .try_into()
+            .unwrap();
+
+        BitArray::new(limbs)
     }
 
     /// Checks if `self` is equal to [Felt::Zero].
-    pub const fn is_zero(&self) -> bool {
-        todo!()
+    pub fn is_zero(&self) -> bool {
+        *self == Felt::ZERO
     }
-
     /// Finite field division.
-    pub const fn field_div(&self, _rhs: &NonZeroFelt) -> Self {
-        todo!()
+    pub fn field_div(&self, rhs: &NonZeroFelt) -> Self {
+        Self(self.0 / rhs.0)
     }
 
     /// Floor division.
-    pub const fn floor_div(&self, _rhs: &NonZeroFelt) -> Self {
-        todo!()
+    pub fn floor_div(&self, rhs: &NonZeroFelt) -> Self {
+        Self(FieldElement::from(
+            &(self.0.representative().div_rem(&rhs.0.representative())).0,
+        ))
     }
 
     /// Multiplicative inverse.
-    pub const fn inverse(&self) -> Option<Self> {
-        todo!()
+    pub fn inverse(&self) -> Option<Self> {
+        Some(Self(self.0.inv()))
     }
 
     /// Finds the square root. There may be 2 roots for each square, and the lower one is returned.
-    pub const fn sqrt(&self) -> Option<Self> {
-        todo!()
+    pub fn sqrt(&self) -> Option<Self> {
+        let (root_1, root_2) = self.0.sqrt()?;
+        Some(Self(core::cmp::min(root_1, root_2)))
     }
 
     /// Raises `self` to the power of 2.
-    pub const fn square(&self) -> Self {
-        todo!()
+    pub fn square(&self) -> Self {
+        Self(self.0.square())
     }
 
     /// Raises `self` to the power of `exponent`.
-    pub const fn pow(&self, _exponent: u128) -> Self {
-        todo!()
+    pub fn pow(&self, exponent: u128) -> Self {
+        Self(self.0.pow(exponent))
     }
 
     /// Modular multiplication.
-    pub const fn mul_mod(&self, _rhs: &Self, _p: &Self) -> Self {
-        todo!()
+    pub fn mul_mod(&self, rhs: &Self, p: &Self) -> Self {
+        Self(FieldElement::from(
+            &(self.0 * rhs.0)
```

Comment:
@xJonathanLEI has a use case for it in [starknet-rs](https://github.com/xJonathanLEI/starknet-rs/blob/ed14f483bb5d5171e6775960a24ee07018f3ff7c/starknet-crypto/src/ecdsa.rs#L88-L118).
I left it as it is part of the initial api, but we could discuss removing it and adding a `mod` method that specifically states that it performs modulo a number smaller than prime to support those use cases. This would also remove the need for `inverse_mod`.

Reply:
I find this confusing and a potential trap. If noone opposed, I say remove.

Reply:
Removed (5fc8a5a)